tests/Unit/TestHelpers.hpp

```diff
@@ -336,3 +337,16 @@ void test_throw_exception(const ThrowingFunctor& func,
     CHECK(false);
   }
 }
+
+/// \ingroup TestingFrameworkGroup
+/// \brief Make a generator of the appropriate type, defaults to `std::mt19937`
+///
+/// \details As the generator is made, `INFO` is called to make sure failed
+/// tests provide seed information.
+template <typename GenType = std::mt19937>
+GenType make_generator() noexcept {
+  std::random_device r;
+  const auto seed = r();
+  INFO("Seed is: " << seed);
```

Comment:
This won't work.  INFO creates an RAII-style object, so it needs to be in a scope enclosing the tests.